### DISCUSSION TITLE
Able to import drawings with .excalidraw format

### DISCUSSION THREAD

**Discussion by Gomilu** [GitbookIO/community#23] [Feature Requests]
### What's your feature/idea?

I would like to import .excalidraw files with the 'drawing' function of gitbook. 
I do have many already drawn .excalidraw files and would like to import them, and continue using there.

### What's your desired outcome?

What would you like to be able to do, if your idea came to life?
Not having to use .svg formats (which are fixed, can't edit) or having to draw a new one

### What's the impact of this for you?

- [ ] Critical
- [ ] High
- [X] Medium
- [ ] Low

### Anything else?

A) 
Exporting as .excalidraw or svg/jpg/png or sth (I am not aware of this image formats) would be nice.

B) 
Found a temporary/inconvenient but doable solution: 

1) open .excalidraw (for example in Obsidian, or on the website of excalidraw) 
2) copy (command A, command C), 
3) open gitbook drawing 
4) paste

**Comment by SamyPesse** [comment]
It's already possible 🎉 

Here is how you can do it:

1. On Excalidraw, save the drawing as an SVG file with the option 'Embed scene" enabled

<img width="1386" alt="Screenshot 2023-01-17 at 17 55 46" src="https://user-images.githubusercontent.com/845425/212962978-a19b0196-08aa-4194-b98a-084547c11fc0.png">

2. Save the file under a name that ends with `.drawing.svg` (not `.excalidraw.svg`) **We should add support for the `.excalidraw.svg` extension as well**

<img width="1386" alt="Screenshot 2023-01-17 at 17 57 14" src="https://user-images.githubusercontent.com/845425/212963228-3ab9953d-f7c3-4029-b41d-c596cacae968.png">

3. In GitBook, insert a drawing block and from the toolbar, click "Replace file"

<img width="1384" alt="Screenshot 2023-01-17 at 17 56 44" src="https://user-images.githubusercontent.com/845425/212963402-3a1caecb-16cf-4cb0-85cb-e9f10e5314c0.png">

4. Select the drawing file from your system

<img width="1374" alt="Screenshot 2023-01-17 at 17 57 29" src="https://user-images.githubusercontent.com/845425/212963484-666bf231-1353-4a7f-8246-c682040fdc15.png">

5. And here we go!
<img width="1375" alt="Screenshot 2023-01-17 at 17 57 38" src="https://user-images.githubusercontent.com/845425/212963514-4941837a-8c09-47b7-aed5-673dde137f70.png">

**Comment by SamyPesse** [comment]
We just released an improvement where the drawing will now use the `.excalidraw.svg` extension.

It means you can now save an image from Excalidraw (with "Embed scene" set to true), and use it as a drawing file in GitBook (using the "Replace file" button on an empty drawing).

**Comment by Gomilu** [comment]
Thank you. I do appreciate your effort a lot!

Good-Intent



Jan 17, 2023, 18:32 by ***@***.***:
